docs/source/CVAOpt_Algorithm/Parameters_and_Hyperparameters.md

```diff
@@ -82,4 +89,89 @@ with TQ42Client() as client:
         compute=HardwareProto.SMALL,
         parameters=params,
     )
-```
\ No newline at end of file
+```
+
+### Example using a channel to evaluate the objective function
+```python
+from tq42.client import TQ42Client
+from tq42.experiment_run import ExperimentRun
+from tq42.organization import list_all as list_all_organizations
+from tq42.project import list_all as list_all_projects
+from tq42.experiment import list_all as list_all_experiments
+from tq42.experiment_run import ExperimentRunStatusProto
+from tq42.algorithm import AlgorithmProto
+from tq42.compute import HardwareProto
+import json
+import pandas
+import asyncio
+import numpy as np
+from tq42.channel import Channel, Ask, Tell
+import OptimizationTestFunctions as otf
+
+cva_params = {}
+cva_params['objectives'] = [{'name': 'Rosenbrock', 'aim_type':'MINIMIZE'}]
+cva_params['variables'] = []
+cva_params['variables'].append({'name': 'x1', 'info_real':{'lower_bound':-1.0, 'upper_bound':1.0}})
+cva_params['variables'].append({'name': 'x2', 'info_real':{'lower_bound':-1.0, 'upper_bound':1.0}})
+cva_params['parameters'] = {}
+cva_params['parameters']['max_generation'] = 50
+cva_params['parameters']['mue'] = 2
+cva_params['parameters']['lambda'] = 10
+
+async def run_exp_with_channel(client, experiment_id, cva_params):
+    # set up channel
+    channel = await Channel.create(client=client)
+    # extend cva_params with func_eval_worker_channel_id
+    cva_params['func_eval_worker_channel_id'] = channel.id
+    
+    # create the experiment run
+    run = ExperimentRun.create(
+        client=client, 
+        algorithm=AlgorithmProto.CVA_OPT, 
+        experiment_id="your_experiment_id",
+        compute=HardwareProto.SMALL,
+        parameters={'parameters': cva_params, 'inputs': {} }
+    )
+    print(f\" starting run with id {run.id}\")
+
+    # define the callback function
+    async def callback(ask: Ask) -> Tell:
+        dim = len(ask.headers)
+        func = otf.Rosenbrock(dim)
+        print('callback received parameters of length: ' + str(len(ask.parameters)))
+        y = []
+        for parameter in ask.parameters:
+            y.append(float(func(np.array(parameter.values))))
+        # add result to data
+
+        return Tell(
+            parameters=ask.parameters,
+            headers=ask.headers,
+            results=y
+        )
+    # define a function to be called after the optimization is finished
+    def success():
+        pass
+
+    # let the channel wait for connections
```

Comment:
not necessary anymore with newest version of SDK. Please look again.
We now have a completion message so you do not need to retry anymore. Just set the timeouts both to `None`. Will run until algo sends completion message.